### PR TITLE
Convert VTTCue to TextTrackCue for rendering captions in Edge

### PR DESCRIPTION
**Pull request by egreaves**
### Changes proposed in this pull request:
Since Edge has support for HLS, the player now uses native rendering for 608 captions tracks. This means that cues from sideloaded tracks need to be of type `TextTrackCue` when  added to a TextTrack.

Fixes #
JW7-2678

### CODE REVIEW

File: src/js/providers/tracks-mixin.js [hunk 1]
```diff
@@ -401,6 +401,18 @@ define(['../utils/underscore',
     ////// PRIVATE METHODS
     //////////////////////
 
+    function _addCueToTrack(track, vttCue) {
+        if (!utils.isEdge()) {
```
Comment (hghazzi) [inline src/js/providers/tracks-mixin.js]: Why did you change this from checking if VTTCue exists to checking if its edge.

Comment (egreaves) [inline src/js/providers/tracks-mixin.js]: @hghazzi `TextTrackCue` also exists in browsers with the `VTTCue` interface, so that wasn't a reliable check. We only need to use `TextTrackCue` in Edge. In IE11, we polyfill `VTTCue` and render with `VTT.js`.

Comment (hghazzi) [inline src/js/providers/tracks-mixin.js]: ok

Comment (robwalch) [inline src/js/providers/tracks-mixin.js]: But you don't know for sure that it exists in all environments that return true for `isEdge` so it would be safer make sure we're not in Edge or don't have TextTrackCues before trying to use them.

`if (!utils.isEdge() || !window.TextTrackCue) {`
